src/app/listSubmissions/ListSubmissionsRequestHandler.ts

```diff
@@ -34,7 +34,12 @@ export class ListSubmissionsRequestHandler {
   }
 
   async handleRequest(parameters: EventParameters): Promise<ApiGatewayV2Response> {
-    const results = await this.database.listSubmissions({ userId: parameters.userId });
+    let results;
+    if (parameters.key) {
+      results = await this.database.getSubmission({ userId: parameters.userId, key: parameters.key });
```

Comment:
Misschien test erbij waarmee je meteen test of je if/else goed ingericht is?
In de test in je mock
```
        getSubmission: async () => {
          return submissionResult;
        },
```
erbij en test erbij?

```
  test('Handler correctly returns getSubmission', async() => {
    const result = await handler.handleRequest({ userId: '900222670', userType: 'person', key: 'submissionkey' });
    expect(result.body).toBe(JSON.stringify(expectedSubmissionResult));
  });
```

Reply:
Goeie, 1-op-1 toegevoegd.